packages/core/src/feed.ts

```diff
@@ -0,0 +1,57 @@
+import { Subject, Observable, map } from 'rxjs'
+import type { StreamMetadata, StreamState } from '@ceramicnetwork/common'
+import { CommitID } from '@ceramicnetwork/streamid'
+
+export type Document = {
+  id: CommitID
+  content: any
+  metadata: StreamMetadata
+}
+
+class DocumentsSubject extends Observable<Document> {
+  readonly subject: Subject<StreamState>
+  constructor() {
+    super((subscriber) => {
+      const subscription = this.subject
+        .pipe(
+          // transform each incoming StreamState to a Document
+          map((streamState: StreamState) => {
+            const tipCID = streamState.log[streamState.log.length - 1].cid
```

Comment:
Do we want any error handling here around these accesses? The log should never be empty, but if we do error here, it might do weird stuff to the observable and the sse.